### DISCUSSION TITLE
Animate icon of NewButtonWithIcon

### DISCUSSION THREAD

**Discussion by aneeskA** [fyne-io/fyne#4942] [General] [answered]
![image](https://github.com/fyne-io/fyne/assets/514419/13774e4e-35c4-4c8c-88c3-6c09c2cb4664)

I am creating a row of icons like this. The first icon when clicked looks for an update from a remote place. So while this is going on, I want to rotate the icon to represent that the action is still ongoing on.

Is there any documentation on how to get this done? Any help will be greatly appreciated.

**Comment by andydotxyz** [comment, marked as answer]
We don't have support for this (neither rotation nor animating icons) so you'll not be able to do it with built-in widgets.
You could however make a custom button and put your own content inside to handle animation I guess.

Or you could manipulate an icon external to the button and constantly change which icon is used by calling `SetIcon`.
